### PR TITLE
Set zbctl user agent

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description
In order to distinguish between go-client and zbctl this PR adds a new possibility to set the UserAgent as client config.
Zbctl use that config and set its own user-agent header.

This is useful to understand which client types and versions are mostly used. See slack https://camunda.slack.com/archives/CHBFBRFM2/p1652420371478469?thread_ts=1651237779.179459&cid=CHBFBRFM2

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

File: clients/go/pkg/zbc/client_test.go [hunk 1]
```diff
@@ -164,6 +166,61 @@ func (s *clientTestSuite) TestGatewayAddressEnvVar() {
 	s.EqualValues(fmt.Sprintf("0.0.0.0:%s", parts[len(parts)-1]), config.GatewayAddress)
 }
 
+func (s *clientTestSuite) TestDefaultUserAgent() {
+	// given
+	var incomingContext = make(map[string][]string)
```
Comment (Zelldon) [inline clients/go/pkg/zbc/client_test.go:171]: :question: Found no better way to verify it. Happy to hear your thoughts

**Review by ChrisKujawa — COMMENTED** [review]
Would be interesting @menski whether you think we should backport this so we have the data also in earlier versions.

**Comment by Zelldon** [comment]
Note: @oleschoenburg the output of https://github.com/camunda/zeebe/runs/6527128287?check_suite_focus=true is not really useful :D do we have this on the radar?

File: clients/go/pkg/zbc/client_test.go (hunk 1 above)
Comment (oleschoenburg) [reply clients/go/pkg/zbc/client_test.go:171]: Other than moving `createServerWithUnaryInterceptor` to some sort of test utility file there's nothing I can think of improving here. We want to test that whoever serves go client requests receives a user-agent header and that that header includes the name and versions so this test looks spot on :+1:

**Review by lenaschoenburg — APPROVED** [review]
Thanks @Zelldon, took me a while to figure out where the version is coming from but I got it now. LGTM :shipit:

**Pull request merged**